xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/transformation/macro/RawBlockFilterParameters.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.transformation.macro;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.xwiki.rendering.transformation.MacroTransformationContext;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Parameters for the {@link RawBlockFilter}.
+ *
+ * @version $Id$
+ * @since 14.7RC1
+ */
+@Unstable
+public class RawBlockFilterParameters
+{
+    /**
+     * If the filtering shall use a safe, restricted mode.
+     */
+    private boolean restricted;
+
+    /**
+     * If cleaning of the input should be performed.
+     */
+    private boolean clean;
+
+    /**
+     * The macro transformation context.
+     */
+    private MacroTransformationContext macroTransformationContext;
```

Comment:
Looking at the code of `HTMLRawBlockFilter` it looks like this field should never be null. So I'd make it final.

Reply:
Yes, though I'm not sure we should really tie this to macro transformations so much. While on the one hand it matches our current use case, we could also extract all properties of the macro transformation we really care about into the parameters and then make the actual filter independent of the macro transformation and thus more generically useful. Depending on what we do with the xdom+xml syntax, we could re-use this filter for example in the xdom+xml parser to filter raw blocks depending on the author. Btw. another use case for this filter (where we have a macro transformation context, though) would be in the context of the RSS macro, where the filtering is currently re-implemented.